src/interpret.jl

```diff
@@ -163,11 +163,25 @@ function evaluate_foreigncall(@nospecialize(recurse), frame::Frame, call_expr::E
     scope = frame.framecode.scope
     data = frame.framedata
     if !isempty(data.sparams) && scope isa Method
-        sig = scope.sig
+        sig = scope.sig    
         args[2] = instantiate_type_in_env(args[2], sig, data.sparams)
-        args[3] = Core.svec(map(args[3]) do arg
-            instantiate_type_in_env(arg, sig, data.sparams)
-        end...)
+        @static if VERSION < v"1.7.0"
+            if args[3] isa Core.SimpleVector
+                args[3] = Core.svec(map(args[3]) do arg
+                    instantiate_type_in_env(arg, sig, data.sparams)
+                end...)
+            else
+                args[3] = instantiate_type_in_env(args[3], sig, data.sparams)
+                args[4] = Core.svec(map(args[4]) do arg
+                    instantiate_type_in_env(arg, sig, data.sparams)
+                end...)
+            end
+        else
+            args[3] = instantiate_type_in_env(args[3], sig, data.sparams)
+            args[4] = Core.svec(map(args[4]) do arg
+                instantiate_type_in_env(arg, sig, data.sparams)
+            end...)
+        end
```

Comment:
```suggestion
        @static if VERSION < v"1.7.0"
            arg3 = args[3]
            if arg3 isa Core.SimpleVector
                args[3] = Core.svec(map(arg3) do arg
                    instantiate_type_in_env(arg, sig, data.sparams)
                end...)
            else
                args[3] = instantiate_type_in_env(arg3, sig, data.sparams)
                args[4] = Core.svec(map(args[4]::Core.SimpleVector) do arg
                    instantiate_type_in_env(arg, sig, data.sparams)
                end...)
            end
        else
            args[3] = instantiate_type_in_env(args[3], sig, data.sparams)
            args[4] = Core.svec(map(args[4]::Core.SimpleVector) do arg
                instantiate_type_in_env(arg, sig, data.sparams)
            end...)
        end
```
for better inferrability.